packages/mui-material/src/List/List.d.ts

```diff
@@ -34,6 +36,11 @@ export interface ListTypeMap<P = {}, D extends React.ElementType = 'ul'> {
      * The system prop that allows defining system overrides as well as additional CSS styles.
      */
     sx?: SxProps<Theme>;
+    /**
+     * The variant to use. There is no variants available by default.
+     * @default 'default'
+     */
+    variant?: ListPropsVariantOverrides;
```

Comment:
Please use `OverridableStringUnion`. You can take a look from `Button.d.ts`.
```suggestion
    variant?: OverridableStringUnion<undefined, ListPropsVariantOverrides>
```

Reply:
> 

That creates a TS error: 
`Type 'undefined' does not satisfy the constraint 'string | number'`

Reply:
I have updated the generator to give an exception to `variant` prop when it is `undefined`.